InnerEye/ML/SSL/lightning_modules/ssl_online_evaluator.py

```diff
@@ -0,0 +1,139 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+import pytorch_lightning as pl
+import torch
+from pl_bolts.callbacks.ssl_online import SSLOnlineEvaluator
+from pl_bolts.models.self_supervised.evaluator import SSLEvaluator
+from torch import Tensor as T
+from torch.nn import functional as F
+
+from InnerEye.ML.SSL.utils import SSLDataModuleType
+from InnerEye.ML.lightning_metrics import Accuracy05, AreaUnderPrecisionRecallCurve, AreaUnderRocCurve, \
+    ScalarMetricsBase
+
+BatchType = Union[Dict[SSLDataModuleType, Any], Any]
+
+
+class SSLOnlineEvaluatorInnerEye(SSLOnlineEvaluator):
+    def __init__(self,
+                 learning_rate: float,
+                 class_weights: Optional[torch.Tensor] = None,
+                 **kwargs: Any) -> None:
+        """
+        Creates a hook to evaluate a linear model on top of an SSL embedding.
+
+        :param class_weights: The class weights to use when computing the cross entropy loss. If set to None,
+                              no weighting will be done.
+        :param length_linear_head_loader: The maximum number of batches in the dataloader for the linear head.
+        """
+
+        super().__init__(**kwargs)
+        self.training_step = int(0)
+        self.weight_decay = 1e-4
+        self.learning_rate = learning_rate
+
+        self.train_metrics: List[ScalarMetricsBase] = [AreaUnderRocCurve(), AreaUnderPrecisionRecallCurve(),
+                                                       Accuracy05()] \
+            if self.num_classes == 2 else [Accuracy05()]
+        self.val_metrics: List[ScalarMetricsBase] = [AreaUnderRocCurve(), AreaUnderPrecisionRecallCurve(),
+                                                     Accuracy05()] \
+            if self.num_classes == 2 else [Accuracy05()]
```

Comment:
What does `accuracy05` mean in the context of multi-class classification? Should we clarify that to avoid confusion, and can the class handle that? I don't have an access to it in this PR.

Reply:
Good point, I think I will create another class that is called multi-class accuracy also to change the name of it.